lib/fs.js

```diff
@@ -732,21 +742,33 @@ fs.truncateSync = function(path, len) {
 };
 
 fs.ftruncate = function(fd, len, callback) {
+  if (!Number.isInteger(fd) || fd < 0)
+    throw new TypeError('fd must be a valid file descriptor');
+
   if (typeof len === 'function') {
     callback = len;
     len = 0;
-  } else if (len === undefined) {
+  } else if (len === undefined || len === null) {
     len = 0;
+  } else if (!Number.isInteger(len) || len < 0) {
+    throw new TypeError('length must be a positive integer');
   }
+
   var req = new FSReqWrap();
   req.oncomplete = makeCallback(callback);
   binding.ftruncate(fd, len, req);
 };
 
 fs.ftruncateSync = function(fd, len) {
-  if (len === undefined) {
+  if (!Number.isInteger(fd) || fd < 0)
```

Comment:
In native it translates fs using `Uint32Value()`. So either we need to check if valid uint32 here, or use `IntegerValue()` on native.

For here the check would look like:
```js
if (fd >>> 0 != fd)
  throw new TypeError(...);
```
Reason for `!=` instead of `!==` is so string values can be properly coerced in C++ using `Uint32Value()`.